Sources/SwiftParser/Declarations.swift

```diff
@@ -1877,12 +1877,55 @@ extension Parser {
   ///     postfix-operator-declaration → 'postfix' 'operator' operator
   ///     infix-operator-declaration → 'infix' 'operator' operator infix-operator-group?
   ///     infix-operator-group → ':' precedence-group-name
+
+  struct OperatorDeclarationIntroducer {
+    var unexpectedBeforeIntroducer: RawUnexpectedNodesSyntax?
+    var operatorModifier: RawTokenSyntax
+    var unexpectedBetweenModifierAndOperatorKeyword: RawUnexpectedNodesSyntax?
+    var operatorKeyword: RawTokenSyntax
+  }
+
+  mutating func parseOperatorDeclarationIntroducer(
+    _ attrs: DeclAttributes,
+    _ handle: RecoveryConsumptionHandle
+  ) -> OperatorDeclarationIntroducer {
+    var modifier = attrs.modifiers?.elements.first?.name ?? self.missingToken(.prefix)
+
+    if ![Keyword.prefix, .infix, .postfix]
+      .map({ $0.defaultText }).contains(modifier.tokenText)
+    {
+      modifier = self.missingToken(.prefix)
+    }
+
+    let (unexpectedBeforeOperator, operatorKeyword) = self.eat(handle)
+
+    let unexpectedBeforeIntroducer = RawUnexpectedNodesSyntax(attrs.attributes?.elements, arena: self.arena)
+
+    var unexpectedBetweenModifierAndOperatorKeyword = unexpectedBeforeOperator
+
+    if modifier.presence == .missing {
+      unexpectedBetweenModifierAndOperatorKeyword = RawUnexpectedNodesSyntax(combining: unexpectedBeforeOperator, RawUnexpectedNodesSyntax(attrs.modifiers?.elements, arena: self.arena), arena: self.arena)
+    } else {
+      unexpectedBetweenModifierAndOperatorKeyword = RawUnexpectedNodesSyntax(combining: unexpectedBeforeOperator, RawUnexpectedNodesSyntax(Array((attrs.modifiers?.elements.dropFirst()) ?? []), arena: self.arena), arena: self.arena)
+    }
+
+    return OperatorDeclarationIntroducer(
+      unexpectedBeforeIntroducer: unexpectedBeforeIntroducer,
+      operatorModifier: modifier,
+      unexpectedBetweenModifierAndOperatorKeyword: unexpectedBetweenModifierAndOperatorKeyword,
+      operatorKeyword: operatorKeyword
+    )
+
+  }
+
   @_spi(RawSyntax)
   public mutating func parseOperatorDeclaration(
     _ attrs: DeclAttributes,
```

Comment:
Instead of retroactively taking `attrs` apart to find the operator, I think it would be easier if `parseOperatorDeclaration` doesn’t take any `attrs` at all. You would just need to make sure that we call into `parseOperatorDeclaration` before the attributes are getting parsed in `parseDeclaration`. I would at least like to see how that looks like. If it’s truly awful, I’m fine with reverting to this design as a last resort.